Libraries/libbom/Headers/bom/bom.h

```diff
@@ -90,6 +93,9 @@ struct bom_tree_context *
 bom_tree_alloc_empty(struct bom_context *context, const char *variable_name);
 
 struct bom_tree_context *
+bom_tree_alloc_size(struct bom_context *context, const char *variable_name, uint32_t index_count);
```

Comment:
This one is a little hard to follow, but in this case the count is also just a shortcut.

The allocated size of the "path indexes" is still recorded as the size of the bom_index saved to tree->child. This is recovered every time the tree is added to via:

   `struct bom_tree_entry *paths = (struct bom_tree_entry *)bom_index_get(tree_context->context, paths_index, &paths_length);`